compiler/rustc_const_eval/src/interpret/util.rs

```diff
@@ -103,5 +103,5 @@ pub(crate) fn create_static_alloc<'tcx>(
     assert_eq!(ecx.machine.static_root_ids, None);
     ecx.machine.static_root_ids = Some((alloc_id, static_def_id));
     assert!(ecx.memory.alloc_map.insert(alloc_id, (MemoryKind::Stack, alloc)).is_none());
-    Ok(ecx.ptr_with_meta_to_mplace(Pointer::from(alloc_id).into(), MemPlaceMeta::None, layout))
+    Ok(ecx.ptr_to_mplace(Pointer::from(alloc_id).into(), layout))
```

Comment:
does this change have anything to do w the pr? it seems unrelated lol

Reply:
oh, i see you changed the privacy of that fn.

Reply:
Yeah, I had to add `ptr_to_mplace_unaligned` for this PR and so I had to make the lower-level `ptr_with_meta_to_mplace` more flexible and didn't want to expose this somewhat "dangerous" operation too far so I checked if I could make it private, and turns out yes I could.